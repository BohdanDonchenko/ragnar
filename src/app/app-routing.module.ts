import { NgModule } from '@angular/core';
import { RouterModule, Routes } from '@angular/router';
import { HomeComponent } from './modules/home/home.component';
import { PageNotFoundComponent } from './modules/shared/components/page-not-found/page-not-found.component';

const routes: Routes = [
  { path: '', redirectTo: '/home', pathMatch: 'full' },
  { path: 'home', component: HomeComponent },
  { path: 'lazy', loadChildren: './modules/lazy/lazy.module#LazyModule' },
  { path: '**', component: PageNotFoundComponent }
];

@NgModule({
  imports: [RouterModule.forRoot(routes)],
  exports: [RouterModule]
})
export class AppRoutingModule {}