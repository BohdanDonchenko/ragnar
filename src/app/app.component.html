<div>
  <h1>Welcome to {{ title }}!</h1>
</div>

<nav class="nav">
  <a class="nav__link a-component" routerLink="/home" routerLinkActive="active">Home</a>
  <a class="nav__link" routerLink="/tasks" routerLinkActive="active">Tasks</a>
</nav>

<router-outlet></router-outlet>
