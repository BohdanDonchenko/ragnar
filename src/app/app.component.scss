@import "../styles/variables";

.a-component {
    color: $a-color-red-n1;
}
