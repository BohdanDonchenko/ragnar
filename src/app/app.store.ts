import { BehaviorSubject } from 'rxjs';
import { Injectable } from '@angular/core';

@Injectable()
export class AppStore {
  userId = new BehaviorSubject<number>(0);
}
