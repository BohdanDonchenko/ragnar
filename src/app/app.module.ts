import { BrowserModule } from '@angular/platform-browser';
import { NgModule, ErrorHandler } from '@angular/core';
import { TasksPageModule } from './pages/tasks/tasks-page.module';
import { HomePageModule } from './pages/home/home-page.module';
import { ServicesModule } from './services/services.module';
import { RepositoriesModule } from './repositories/repositories.module';

import { AppRoutingModule } from './app-routing.module';
import { AppComponent } from './app.component';
import { GlobalErrorHandler } from './infrastructure/global-error-handler';
import { AppStore } from './app.store';

@NgModule({
  declarations: [
    AppComponent,
  ],
  imports: [
    BrowserModule,
    AppRoutingModule,
    RepositoriesModule,
    ServicesModule,
    HomePageModule,
    TasksPageModule
  ],
  providers: [
    {
      provide: ErrorHandler,
      useClass: GlobalErrorHandler
    },
    AppStore
  ],
  bootstrap: [AppComponent],
})
export class AppModule { }
