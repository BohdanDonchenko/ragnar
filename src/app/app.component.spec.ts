import { TestBed, async } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { AppComponent } from './app.component';

describe('AppComponent', () => {
  describe('Snapshot', () => {
    beforeEach(async(() => {
      TestBed.configureTestingModule({
        imports: [RouterTestingModule],
        declarations: [AppComponent],
      }).compileComponents();
    }));

    test('check snapshot', () => {
      const sut = TestBed.createComponent(AppComponent);

      expect(sut).toMatchSnapshot();
    });
  });

  describe('Functionality', () => {
    test('constructor', () => {
      const sut = new AppComponent();
      expect(sut.title).toEqual('Ragnar');
    });
  });
});
