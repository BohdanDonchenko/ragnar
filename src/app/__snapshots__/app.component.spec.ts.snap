// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent Snapshot check snapshot 1`] = `
<app-root
  title={[Function String]}
>
  <div>
    <h1>
      Welcome to 
    </h1>
  </div><nav
    class="nav"
  >
    <a
      class="nav__link a-component"
      routerlink="/home"
      routerlinkactive="active"
    >
      Home
    </a>
    <a
      class="nav__link"
      routerlink="/tasks"
      routerlinkactive="active"
    >
      Tasks
    </a>
  </nav><router-outlet />
</app-root>
`;
